src/hato/middleware.clj

```diff
@@ -695,10 +697,18 @@
   "Adds appropriate body and header if making a multipart request."
   [{:keys [multipart] :as req}]
   (if multipart
-    (let [b (multipart/boundary)]
+    (let [b (multipart/boundary)
+          segments (multipart/raw-multipart-payload-segments multipart b)
+          input-stream (multipart/body segments)
+          content-length (multipart/content-length segments)]
       (-> req
           (dissoc :multipart)
-          (assoc :body (multipart/body multipart b))
+          (assoc :body
+                 (if (= -1 content-length)
```

Comment:
If we are unable to determine the content length, leave the body as an input stream and use transfer encoding chunked. Otherwise, specify the content length so the appropriate content-length header is used.